pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java

```diff
@@ -543,70 +540,83 @@ protected void setupCustomRouting(@NonNull Config cfg) {
         String auth0Domain = cfg.getString(ApplicationConfigConstants.AUTH0_DOMAIN);
         String auth0claimNameSpace = cfg.getString(ApplicationConfigConstants.AUTH0_CLAIM_NAMESPACE);
 
         if (StringUtils.isBlank(bspSecret)) {
-            throw new RuntimeException("No secret supplied for BSP endpoint, system exiting.");
+            throw new DsmInternalError("No secret supplied for BSP endpoint, system exiting.");
         }
 
         if (StringUtils.isBlank(ddpSecret)) {
-            throw new RuntimeException("No secret supplied for DDP endpoint, system exiting.");
+            throw new DsmInternalError("No secret supplied for DDP endpoint, system exiting.");
         }
 
         String appRoute = cfg.hasPath("portal.appRoute") ? cfg.getString("portal.appRoute") : null;
 
         if (StringUtils.isBlank(appRoute)) {
-            throw new RuntimeException("appRoute was not configured correctly.");
+            throw new DsmInternalError("appRoute was not configured correctly.");
         }
 
-        //  capture basic route info for logging
-        //        before("*", new LoggingFilter(auth0Domain, auth0claimNameSpace));
-        before(apiRoot + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, bspSecret, null, bspSecretEncoded));
-        before(uiRoot + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, null, null, false));
-        before(infoRoot + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, signer, ddpSecretEncoded));
-        before(appRoute + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, signer, ddpSecretEncoded));
+        before(API_ROOT + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, bspSecret, null, bspSecretEncoded));
+        before(DSM_ROOT + "*",
+                new LoggingFilter(auth0Domain, auth0claimNameSpace, dsmSecretForJuniper, juniperSigner, juniperSecretEncoded));
+        before(UI_ROOT + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, null, null, false));
+        before(INFO_ROOT + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, KDUX_SIGNER, ddpSecretEncoded));
+        before(appRoute + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, KDUX_SIGNER, ddpSecretEncoded));
         afterAfter((req, res) -> MDC.clear());
 
-        before(apiRoot + "*", (req, res) -> {
+        before(API_ROOT + "*", (req, res) -> {
             if (!new JWTRouteFilter(auth0Domain).isAccessAllowed(req, false, bspSecret)) {
                 logger.info("Returning 401 because token was not verified");
                 halt(401);
             }
             res.header(HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString());
         });
 
+        before(DSM_ROOT + "*", (req, res) -> {
+            if (!new JWTRouteFilter(auth0Domain).isAccessAllowed(req, false, dsmSecretForJuniper)) {
+                logger.info("Returning 401 because token was not verified");
+                halt(401);
```

Comment:
It would be helpful to include a message with the 401 (e.g. "Token could not be verified").